engine/events/specials.asm

```diff
@@ -354,54 +354,78 @@ RespawnOneOffs:
 	eventflagreset EVENT_BEAT_CYNTHIA
 .SkipCynthia
 
-	ld a, SUDOWOODO - 1
-	call CheckCaughtMon
+	ld de, ENGINE_PLAYER_CAUGHT_SUDOWOODO
+	ld b, CHECK_FLAG
+	call _EngineFlagAction
+	ld a, c
+	and a
```

Comment:
A lot of redundant code here... Not sure if I should optimize for cycles or bytes here.

Optimizing for bytes would be something like:
```asm
	ld de, ENGINE_PLAYERCAUGHT_SUDOWOODO
	call CheckCaughtFlag
	jr nz, .CaughtSudowoodo
	eventflagreset EVENT_ROUTE_36_SUDOWOODO
```

```asm
CheckCaughtFlag:
	ld b, CHECK_FLAG
	call _EngineFlagAction
	ld a, c
	and a
	ret
```
I'm estimating this would save roughly 33 bytes... but add an extra 110 cycles.

Reply:
The whole routine is doing a lot of slow flag work anyway, so may as well optimize for size.